### DISCUSSION TITLE
Implementation of custom hulls

### DISCUSSION THREAD

**Discussion by mankrip** [ericwa/ericw-tools#341] [Ideas]
Idea: Generate a custom last hull in Q1BSP files if the fields "hull3_mins" and "hull3_maxs" are present in worldspawn.

This way,
1) Custom engines will be able to detect it and assign the proper size;
2) Engines that doesn't support it won't crash (they simply ignore the last hull, as seen in Mod_LoadClipnodes and SV_HullForEntity);
3) The user will be warned if the engine doesn't support it ("hull3_mins is not a field" message), as long as those fields aren't present in the qc code.

It's a simple solution that requires minimal changes to both the engine and qbsp, and would allow modders to implement true crouching.

There's also potential for customizing hulls 1 and 2 (hull 0 will always be point-sized), but I don't think it's needed yet.

**Comment by ericwa** [comment]
This sounds fine

**Comment by ericwa** [comment]
“numhulls” could also fit into this, ie the Quake default would be 3.
